DatadogSessionReplay/Sources/Writers/ResourcesWriter.swift

```diff
@@ -16,25 +16,98 @@ internal protocol ResourcesWriting {
 
 internal class ResourcesWriter: ResourcesWriting {
     /// An instance of SDK core the SR feature is registered to.
-    private weak var core: DatadogCoreProtocol?
+    private let scope: FeatureScope?
+    private let telemetry: Telemetry
+
+    @ReadWriteLock
+    private var knownIdentifiers = Set<String>() {
+        didSet {
+            if let knownIdentifiers = knownIdentifiers.asData() {
+                scope?.dataStore.setValue(
+                    knownIdentifiers,
+                    forKey: Constants.knownResourcesKey
+                )
+            }
+        }
+    }
 
     init(
-        core: DatadogCoreProtocol
+        core: DatadogCoreProtocol,
+        dataStoreResetTime: TimeInterval = TimeInterval(30).days
     ) {
-        self.core = core
+        self.scope = core.scope(for: ResourcesFeature.self)
+        self.telemetry = core.telemetry
+
+        self.scope?.dataStore.value(forKey: Constants.storeCreationKey) { result in
+            if let storeCreation = result.data()?.asTimeInterval(), Date().timeIntervalSince1970 - storeCreation < dataStoreResetTime {
+                self.scope?.dataStore.value(forKey: Constants.knownResourcesKey) { [weak self] result in
+                    switch result {
+                    case .value(let data, _):
+                        if let knownIdentifiers = data.asKnownIdentifiers() {
+                            self?.knownIdentifiers.formUnion(knownIdentifiers)
+                        }
+                    case .error(let error):
+                        self?.telemetry.error("Failed to read processed resources from data store: \(error)")
```

Comment:
suggestion/ The DS implementation [already sends](https://github.com/DataDog/dd-sdk-ios/blob/d8f1ea77a8db454f0b1c3814d30831d10551582d/DatadogCore/Sources/Core/DataStore/FeatureDataStore.swift#L69-L70) telemetry for `.error()` case, there's no reason for double reporting it.
```suggestion
```